packages/shared/src/mock/values.ts

```diff
@@ -1,189 +1,241 @@
-  float32_rgb: Float32Array.from(rgbFlat.map((val) => val / 255)),
-  X: arr1,
-  X_desc: [...arr1].reverse(),
-  X_log: arr1.map((_, i) => (i + 1) * 0.1),
-  X_rgb: arr5,
-  Y: arr2,
-  Y_desc: [...arr2].reverse(),
-  Y_rgb: arr6,
-  title_twoD: 'NeXus 2D',
-  secondary: twoD.map((inner) => inner.map((v) => v * 2)),
-  tertiary: twoD.map((inner) => inner.map((v) => v / 2)),
-  position: [-1, 1, 3], // pixel boundaries (N + 1)
-  scatter_data: arr1.map((val) => Math.cos((val * 3.14) / 40)),
-  Y_scatter: arr1.map((_, i) => ((i % 10) + (i % 5)) / 123_456),
-  oneD_compound,
-  twoD_compound,
-};
+  threeD_bool: () =>
+    ndarray(
+      [
+        [
+          [true, false, true, false],
+          [true, true, true, true],
+          [false, false, false, false],
+        ],
+        [
+          [true, false, false, false],
+          [true, true, true, true],
+          [false, false, true, false],
+        ],
+      ].flat(2),
+      [2, 3, 4],
+    ),
+  threeD_rgb,
+  fourD: () => {
+    const { data: dataThreeD } = threeD();
+    return ndarray(
+      range4().flatMap((divider) =>
+        dataThreeD.map((val) => Math.sin(val / divider)),
+      ),
+      [3, 9, 20, 41],
+    );
+  },
+  fourD_rgb: () => {
+    const { data: dataRgb } = threeD_rgb();
+    return ndarray(
+      range8().flatMap((i) =>
+        dataRgb.map((val) => Math.min(val + i * 20, 255)),
+      ),
+      [10, 3, 3, 3],
+    );
+  },
+  uint8: () => ndarray(Uint8Array.from(range7()), [2, 2]),
+  int16: () => ndarray(Int16Array.from(range7()), [2, 2]),
+  float32: () => ndarray(Float32Array.from(range7()), [2, 2]),
+  float64: () => ndarray(Float64Array.from(range7()), [2, 2]),
+  int8_rgb: () =>
+    ndarray(
+      Int8Array.from(threeD_rgb().data.map((val) => val - 128)),
+      [3, 3, 3],
+    ),
+  uint8_rgb: () => ndarray(Uint8Array.from(threeD_rgb().data), [3, 3, 3]),
+  int32_rgb: () => ndarray(Uint32Array.from(threeD_rgb().data), [3, 3, 3]),
+  float32_rgb: () =>
+    ndarray(
+      Float32Array.from(threeD_rgb().data.map((val) => val / 255)),
+      [3, 3, 3],
+    ),
+  X: () => ndarray(range1()),
+  X_desc: () => {
+    const arr = range1();
+    arr.reverse();
+    return ndarray(arr);
+  },
+  X_log: () => ndarray(range1().map((_, i) => (i + 1) * 0.1)),
+  X_rgb: () => ndarray(range5()),
+  Y: () => ndarray(range2()),
+  Y_desc: () => {
+    const arr = range1();
+    arr.reverse();
+    return ndarray(arr);
+  },
+  Y_rgb: () => ndarray(range6()),
+  secondary: () =>
+    ndarray(
+      twoD().data.map((v) => v * 2),
+      [20, 41],
+    ),
+  tertiary: () =>
+    ndarray(
+      twoD().data.map((v) => v / 2),
+      [20, 41],
+    ),
+  position: () => ndarray([-1, 1, 3]), // pixel boundaries (N + 1)
+  scatter_data: () =>
+    ndarray(range1().map((val) => Math.cos((val * 3.14) / 40))),
+  Y_scatter: () =>
+    ndarray(range1().map((_, i) => ((i % 10) + (i % 5)) / 123_456)),
+} satisfies Record<string, () => NdArray<ArrayValue<DType>>>;
```

Comment:
Very important: this helps make sure `mockValues` contains only functions and that they all return ndarrays. If `mockValues` were to contain anything else, then we'd get a complicated type error in the `array()` function.